This patch allows quarto-cli to build on Deno 1.31.
Deno.build.os now has new options.
If you use one of these OSes, you probably should edit this patch and
grep for Deno.build.os to expand on this patch.

--- a/src/core/appdirs.ts
+++ b/src/core/appdirs.ts
@@ -43,6 +43,8 @@ export function userDataDir(appName: string, roaming = false) {
       return xdgUserDataDir(appName);
     case "windows":
       return windowsUserDataDir(appName, roaming);
+    default:
+      return "";
   }
 }
 
@@ -54,6 +56,8 @@ export function userConfigDir(appName: string, roaming = false) {
       return xdgUserConfigDir(appName);
     case "windows":
       return windowsUserDataDir(appName, roaming);
+    default:
+      return "";
   }
 }
 
@@ -65,6 +69,8 @@ export function userCacheDir(appName: string) {
       return xdgUserCacheDir(appName);
     case "windows":
       return windowsUserDataDir(appName);
+    default:
+      return "";
   }
 }
 
@@ -76,6 +82,8 @@ export function userRuntimeDir(appName: string) {
       return xdgUserRuntimeDir(appName);
     case "windows":
       return windowsUserDataDir(appName);
+    default:
+      return "";
   }
 }
 
--- a/src/core/shell.ts
+++ b/src/core/shell.ts
@@ -14,6 +14,11 @@ export async function openUrl(url: string) {
     windows: "explorer",
     darwin: "open",
     linux: "xdg-open",
+    freebsd: "",
+    netbsd: "",
+    aix: "",
+    solaris: "",
+    illumos: "",
   };
 
   const cmd = shellOpen[Deno.build.os];
